### ISSUE TITLE
Mac: "Full Keyboard Access" renders Zed inoperable.

### ISSUE THREAD

**Issue by ThomasHaz** [zed-industries/zed#14390]
### Summary

On macOS Sonoma (14.5), enabling:
`System Settings > Accessibility > Keyboard > Full Keyboard Access`

Disables the space key, both in the editor and the terminal.

It may be to do with them not receiving, or retaining, focus?

### Environment

Zed: v0.143.7 (Zed)
OS: macOS 14.5.0
Memory: 16 GiB
Architecture: aarch64

**Comment by notpeter** [comment]
Yep, you can quickly reproduce by triple tapping the TouchID button and toggling "Full Keyboard Access".  Space key doesn't work and MacOS will highlight a control in the last application which was previously selected and draw an outline that covers Zed.

(camera phone shot because the blue outline is hidden from screenshots)
![IMG_9546](https://github.com/user-attachments/assets/16f564d0-cf49-463b-9002-13b735e7023c)

**Comment by dionisvl** [comment]
up

**Comment by notpeter** [comment]
not stale

**Comment by cacaosteve** [comment]
@notpeter or @ThomasHaz would you like to take a look at or try my fix? https://github.com/zed-industries/zed/pull/57579
